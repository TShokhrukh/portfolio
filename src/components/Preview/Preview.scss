.Preview {
  --preview-background: center / contain no-repeat url('./assets/preview-bg.svg');
}
