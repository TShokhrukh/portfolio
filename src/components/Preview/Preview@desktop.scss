.Preview_desktop  {
  background: var(--preview-background);

  .Preview-Container {
    display: flex;
    justify-content: center;
    flex-direction: column;
    height: 100%;
  }

  .Preview-Content {
    display: flex;
    align-items: center;
    justify-content: space-between;
  }
}

