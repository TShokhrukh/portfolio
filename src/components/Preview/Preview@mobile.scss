.Preview_mobile  {
  .Preview-Background {
    background: var(--preview-background);
  }

  .Preview-Container {
    height: 100%;
    display: grid;
    grid-template-rows: auto 1fr auto;
  }

  .Preview-Social {
    display: flex;
    align-items: center;
    justify-content: center;
  }
}
