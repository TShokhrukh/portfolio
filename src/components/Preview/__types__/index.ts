import type { ElementType, ReactNode } from 'react'
import type { IClassNameProps } from '@bem-react/core'

export interface IPreviewProps extends IClassNameProps {
  component?: ElementType
}

export interface IPreviewIconProps extends IClassNameProps {
  name: string
}

export interface IPreviewContentProps extends IClassNameProps {
  children?: ReactNode,
}
