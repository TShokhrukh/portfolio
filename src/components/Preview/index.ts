export * from './build'
export * from './__types__'
export { Preview as PreviewBase } from './Preview'
