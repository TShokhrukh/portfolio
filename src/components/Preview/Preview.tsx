import React from 'react'
import { cnPreview } from './Preview.const'
import PreviewSocial from './Social'
import PreviewContainer from './Container'
import PreviewAbout from './About'
import PreviewCV from './CV'
import PreviewContent from './Content'

import type { IPreviewProps } from './__types__'

import './Preview.scss'

export const Preview: React.FC<IPreviewProps> = ({
  component: Component = 'div',
  className,
  ...props
}) => {
  return (
    <Component {...props} className={cnPreview({}, [className])}>
      <div className='Preview_mobile'>
        <PreviewContainer >
          <PreviewSocial />
          <div className='Preview_bg' />
          <div>
            <PreviewAbout />
            <PreviewCV />
          </div>
        </PreviewContainer>
      </div>
      <div className='Preview_desktop Preview_bg'>
        <PreviewContainer>
          <PreviewContent className={cnPreview('Content')}>
            <div>
              <PreviewAbout />
              <PreviewSocial />
            </div>
            <PreviewCV />
          </PreviewContent>
        </PreviewContainer>
      </div>
    </Component>
  )
}
