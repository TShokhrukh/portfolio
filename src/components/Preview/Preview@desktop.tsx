import React from 'react'
import { cnPreview } from './Preview.const'
import PreviewSocial from './Social'
import PreviewContainer from './Container'
import PreviewAbout from './About'
import PreviewCV from './CV'
import PreviewContent from './Content'

import type { IPreviewProps } from './__types__'

import './Preview.scss'
import './Preview@desktop.scss'

export const Preview: React.FC<IPreviewProps> = ({
  component: Component = 'div',
  className,
  ...props
}) => {
  return (
    <Component {...props} className={cnPreview({ desktop: true }, [className])}>
      <PreviewContainer>
        <PreviewContent>
          <div>
            <PreviewAbout />
            <PreviewSocial />
          </div>
          <PreviewCV />
        </PreviewContent>
      </PreviewContainer>
    </Component>
  )
}
