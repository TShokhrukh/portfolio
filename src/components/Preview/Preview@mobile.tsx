import React from 'react'
import { cnPreview } from './Preview.const'
import PreviewSocial from './Social'
import PreviewContainer from './Container'
import PreviewAbout from './About'
import PreviewCV from './CV'

import type { IPreviewProps } from './__types__'

import './Preview.scss'
import './Preview@mobile.scss'

export const Preview: React.FC<IPreviewProps> = ({
  component: Component = 'div',
  className,
  ...props
}) => {
  return (
    <Component {...props} className={cnPreview({ mobile: true }, [className])}>
      <PreviewContainer>
        <PreviewSocial />
        <div className={cnPreview('Background')} />
        <div>
          <PreviewAbout />
          <PreviewCV />
        </div>
      </PreviewContainer>
    </Component>
  )
}
