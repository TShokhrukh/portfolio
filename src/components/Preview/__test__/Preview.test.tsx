/* global test, expect, describe */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { Preview as PreviewBase } from '../Preview'
import { Preview as PreviewMobile } from '../Preview@mobile'
import { Preview as PreviewDesktop } from '../Preview@desktop'
import { PreviewWrapper } from './PreviewWrapper'

const platforms = [
  ['base', PreviewBase],
  ['mobile', PreviewMobile],
  ['desktop', PreviewDesktop]
]

describe.each(platforms)('Preview@%s', (_, Preview) => {
  test('must return the component template', () => {
    render(
      <PreviewWrapper>
        <Preview className='test_class' data-testid='preview' />
      </PreviewWrapper>
    )

    expect(screen.getByTestId('preview')).toMatchSnapshot()
  })

  test('must change the component tag via prop component', () => {
    render(
      <PreviewWrapper>
        <Preview component='section' data-testid='preview' />
      </PreviewWrapper>
    )

    expect(screen.getByTestId('preview').tagName).toBe('SECTION')
  })
})
