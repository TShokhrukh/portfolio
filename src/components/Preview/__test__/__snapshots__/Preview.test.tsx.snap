// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview@base must return the component template 1`] = `
<div
  class="Preview test_class"
  data-testid="preview"
>
  <div
    class="Preview_mobile"
  >
    <div
      class="Container Preview-Container"
    >
      <div
        class="Preview-Social"
      >
        <a
          class="Link Preview-SocialLink"
          href="https://github.com/TShokhrukh"
          target="_black"
        >
          <span
            aria-hidden="true"
            class="fa fa-github Icon"
          />
        </a>
      </div>
      <div
        class="Preview_bg"
      />
      <div>
        <div
          class="Preview-About"
        >
          <span
            class="Badge Badge_color_purple Preview-AboutBadge"
          >
            preview.greeting
          </span>
          <h1
            class="Preview-AboutTitle"
          >
            <span>
              preview.name
            </span>
          </h1>
          <p
            class="Preview-AboutDesc"
          >
            <span>
              preview.skills
            </span>
          </p>
        </div>
        <div
          class="Preview-CV"
        >
          <a
            class="Link Preview-CVLink"
            href="#adw"
            target="_black"
          >
            <span>
              preview.downloadCv
            </span>
            <span
              aria-hidden="true"
              class="fa fa-download Icon Link-Icon Link-Icon_direction_end"
            />
          </a>
          <button
            class="Button"
            type="button"
          >
            <span
              aria-hidden="true"
              class="fa fa-telegram Icon Button-Icon Button-Icon_direction_start"
            />
            <span>
              preview.talk
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    class="Preview_desktop Preview_bg"
  >
    <div
      class="Container Preview-Container"
    >
      <div
        class="Preview-Content"
      >
        <div>
          <div
            class="Preview-About"
          >
            <span
              class="Badge Badge_color_purple Preview-AboutBadge"
            >
              preview.greeting
            </span>
            <h1
              class="Preview-AboutTitle"
            >
              <span>
                preview.name
              </span>
            </h1>
            <p
              class="Preview-AboutDesc"
            >
              <span>
                preview.skills
              </span>
            </p>
          </div>
          <div
            class="Preview-Social"
          >
            <a
              class="Link Preview-SocialLink"
              href="https://github.com/TShokhrukh"
              target="_black"
            >
              <span
                aria-hidden="true"
                class="fa fa-github Icon"
              />
            </a>
          </div>
        </div>
        <div
          class="Preview-CV"
        >
          <a
            class="Link Preview-CVLink"
            href="#adw"
            target="_black"
          >
            <span>
              preview.downloadCv
            </span>
            <span
              aria-hidden="true"
              class="fa fa-download Icon Link-Icon Link-Icon_direction_end"
            />
          </a>
          <button
            class="Button"
            type="button"
          >
            <span
              aria-hidden="true"
              class="fa fa-telegram Icon Button-Icon Button-Icon_direction_start"
            />
            <span>
              preview.talk
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Preview@desktop must return the component template 1`] = `
<div
  class="Preview Preview_desktop test_class"
  data-testid="preview"
>
  <div
    class="Container Preview-Container"
  >
    <div
      class="Preview-Content"
    >
      <div>
        <div
          class="Preview-About"
        >
          <span
            class="Badge Badge_color_purple Preview-AboutBadge"
          >
            preview.greeting
          </span>
          <h1
            class="Preview-AboutTitle"
          >
            <span>
              preview.name
            </span>
          </h1>
          <p
            class="Preview-AboutDesc"
          >
            <span>
              preview.skills
            </span>
          </p>
        </div>
        <div
          class="Preview-Social"
        >
          <a
            class="Link Preview-SocialLink"
            href="https://github.com/TShokhrukh"
            target="_black"
          >
            <span
              aria-hidden="true"
              class="fa fa-github Icon"
            />
          </a>
        </div>
      </div>
      <div
        class="Preview-CV"
      >
        <a
          class="Link Preview-CVLink"
          href="#adw"
          target="_black"
        >
          <span>
            preview.downloadCv
          </span>
          <span
            aria-hidden="true"
            class="fa fa-download Icon Link-Icon Link-Icon_direction_end"
          />
        </a>
        <button
          class="Button"
          type="button"
        >
          <span
            aria-hidden="true"
            class="fa fa-telegram Icon Button-Icon Button-Icon_direction_start"
          />
          <span>
            preview.talk
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Preview@mobile must return the component template 1`] = `
<div
  class="Preview Preview_mobile test_class"
  data-testid="preview"
>
  <div
    class="Container Preview-Container"
  >
    <div
      class="Preview-Social"
    >
      <a
        class="Link Preview-SocialLink"
        href="https://github.com/TShokhrukh"
        target="_black"
      >
        <span
          aria-hidden="true"
          class="fa fa-github Icon"
        />
      </a>
    </div>
    <div
      class="Preview-Background"
    />
    <div>
      <div
        class="Preview-About"
      >
        <span
          class="Badge Badge_color_purple Preview-AboutBadge"
        >
          preview.greeting
        </span>
        <h1
          class="Preview-AboutTitle"
        >
          <span>
            preview.name
          </span>
        </h1>
        <p
          class="Preview-AboutDesc"
        >
          <span>
            preview.skills
          </span>
        </p>
      </div>
      <div
        class="Preview-CV"
      >
        <a
          class="Link Preview-CVLink"
          href="#adw"
          target="_black"
        >
          <span>
            preview.downloadCv
          </span>
          <span
            aria-hidden="true"
            class="fa fa-download Icon Link-Icon Link-Icon_direction_end"
          />
        </a>
        <button
          class="Button"
          type="button"
        >
          <span
            aria-hidden="true"
            class="fa fa-telegram Icon Button-Icon Button-Icon_direction_start"
          />
          <span>
            preview.talk
          </span>
        </button>
      </div>
    </div>
  </div>
</div>
`;
